arch/risc-v/include/arch.h

```diff
@@ -60,6 +60,12 @@ static inline uintptr_t up_getsp(void)
   return sp;
 }
 
+/* The kernel needs this in case CONFIG_SMP is defined */
+
+#ifdef CONFIG_SMP
+#define up_cpu_index (int)riscv_cpuindex
```

Comment:
I'm not sure if this is a correct way since in `include/nuttx/arch.h` I see
```
#ifdef CONFIG_SMP
int up_cpu_index(void);
#else
#  define up_cpu_index() (0)
#endif
```
so now compilation is include order dependent. If `include/nuttx/arch.h` is included after this header, then compilation will fail and this is not a good.
Maybe better to modify the `int up_cpu_index(void);` -> `size_t up_cpu_index(void);` and do all the necessary changes in all places inside the project?
@xiaoxiang781216 @no1wudi @masayuki2009 what do you think?